pennylane/pytrees/pytrees.py

```diff
@@ -217,11 +217,18 @@ def __str__(self):
 leaf = PyTreeStructure(None, (), [])
 
 
-def flatten(obj: Any) -> tuple[list[Any], PyTreeStructure]:
+def flatten(
+    obj: Any, is_leaf: Callable[[Any], bool] | None = None
```

Comment:
```suggestion
    obj: Any, is_leaf: Optional[Callable[[Any], bool]] = None
```
Note that we still support Python 3.9, at least for the near future.  The `| None` syntax is only available in Python 3.10+.